client/packages/inventory/src/Stocktake/DetailView/modal/StocktakeLineEdit/StocktakeLineEditTables.tsx

```diff
@@ -104,14 +106,19 @@ const getInventoryAdjustmentReasonInputColumn = (
         rowData.snapshotNumberOfPacks -
         (rowData.countedNumberOfPacks || rowData.snapshotNumberOfPacks);
 
+      const errorType = getError(rowData)?.__typename;
+      const isAdjustmentReasonError =
+        errorType === 'AdjustmentReasonNotProvided' ||
+        errorType === 'AdjustmentReasonNotValid';
+
       return (
         <InventoryAdjustmentReasonSearchInput
```

Comment:
While looking at this PR i was tempted to refactor this, I didn't see a reason why `InventoryAdjustmentReasonSearchInput` need to be a common component, and why we would need the wrapper `getInventoryAdjustmentReasonInputColumn` for it, rather then using `Cell` property and   `InventoryAdjustmentReasonSearchInput` defined in inventory package, and `InventoryAdjustmentReasonSearchInput` can have all of the logic.

Reply:
yes, could do. Another issue though please! this PR is already a long one to review 😉

Reply:
Can we leave it for now and maybe do the change if we touching the area ? At least we are of a similar opinion on this front, I've left a comment:

94e3ae2b1d270cc21f62dc293e8e3845481ab03f